### ISSUE TITLE
Fails to load projects

### ISSUE THREAD

**Issue by wgrs** [microsoft/vscode-dotnettools#59]
### Describe the Issue

After initialization, all projects fail to load
2023-06-07 22:56:52.240 [info] Project system initialization finished. 0 project(s) are loaded, and 6 failed to load.

### Steps To Reproduce

1. Add the extension
2. Close and Open VSCode
3. Projects fail to load.

### Expected Behavior

All projects load.

### Environment Information

Linux Kubuntu 22.04
ms-dotnettools.csdevkit
v0.1.83

**Comment by wgrs** [comment]
> Is it also giving you "A compatible .NET SDK was not found"? I'm on Alpine, seeing the same thing, both on .NET 6 and 7.

No, I am not getting that error now, but I did have that issue on Linux when I first installed the SDK - After multiple re installs, it eventually corrected itself.

**Comment by timheuer** [comment]
@wgrs you mention "Kubuntu" here -- is that explicit? I doubt there are huge differences here, but just a note that it is not a distro  that VS Code supports.

Outside of that is there more info you can provide? is the project available in public? or can you provide the logs (from command pallette - 'open extensions logs') and zip up the one for the csdevkit?

**Comment by wgrs** [comment]
> @wgrs you mention "Kubuntu" here -- is that explicit? I doubt there are huge differences here, but just a note that it is not a distro that VS Code supports.
> 
> Outside of that is there more info you can provide? is the project available in public? or can you provide the logs (from command pallette - 'open extensions logs') and zip up the one for the csdevkit?

Kububtu is Ubuntu - Its a supported distro.

If you dotnet new webapi -n testext , you will see it fail. Brand new clean project fails.

**Comment by wgrs** [comment]
[exthost.zip](https://github.com/microsoft/vscode-dotnettools/files/11684141/exthost.zip)

Let me know if these are the correct logs

**Comment by wgrs** [comment]
> @wgrs you mention "Kubuntu" here -- is that explicit? I doubt there are huge differences here, but just a note that it is not a distro that VS Code supports.
> 
> Outside of that is there more info you can provide? is the project available in public? or can you provide the logs (from command pallette - 'open extensions logs') and zip up the one for the csdevkit?

[exthost.zip](https://github.com/microsoft/vscode-dotnettools/files/11684142/exthost.zip)

**Comment by vnv-dwarf** [comment]
I also have this problem on windows where the project could not load because there is a space in the path (my user folder has a space in it)
```
2023-06-08 09:01:51.420 [info] Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet-c:\Users\Nicolas does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
```

**Comment by TheRancid** [comment]
Confirming the same bug on my Windows machine, having a space in my profile folder too 😞 

> I also have this problem on windows where the project could not load because there is a space in the path (my user folder has a space in it)
> 
> ```
> 2023-06-08 09:01:51.420 [info] Could not execute because the specified command or file was not found.
> Possible reasons for this include:
>   * You misspelled a built-in dotnet command.
>   * You intended to execute a .NET program, but dotnet-c:\Users\Nicolas does not exist.
>   * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
> ```

**Comment by ryandeering** [comment]
Getting the same on macOS Ventura.

I also get this under C# Dev Kit - Test Explorer:

```
Created Test Controller
Using vstest from dotnet sdk in [/usr/local/share/dotnet/sdk/6.0.407].
System.Exception: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/6.0.407/vstest.console.dll" /parentprocessid:14107 /port:50800.
 ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process '/usr/local/bin/dotnet' with working directory '/Users/ryand/Desktop/app'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   --- End of inner exception stack trace ---
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProviders.LocalTestPlatformProvider.ActivateSessionAsync(ITestWindowCallbackService testWindowCallbackService, ITestAgentBridgeCallback TestAgentBridgeCallback, Boolean enableDiagnosticLogging, RuntimeEnvironment runtimeEnvironment, IDictionary`2 environment, IDictionary`2 props, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformAgentPool.<>c__DisplayClass24_0.<<CreateRemoteAgentAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Utilities.EventPumpExtensions.<>c__DisplayClass4_0`1.<<EnqueueAsync>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Extensibility.TaskCompletionSourceExtensions.RunTaskWithCatchAsync[T](TaskCompletionSource`1 taskCompletionSource, Func`1 task)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProvider.GetProviderAsync(ITestWindowCallbackService callbackClient, IInternalLogger logger, ITestWindowHostTelemetry telemetryService, ICollection`1 runtimeEnvironments, IDictionary`2 environmentVariables, Boolean enableDiagnosticLogging, Boolean autoInstallRuntimePrereqs, ITestPlatformProviderMap testPlatformProviderMap)
   at Microsoft.VisualStudio.TestWindow.Host.TestWindowStoreService.InitializeVSCodeServiceAsync(InitializeOptions initializeRequest, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.VSCode.Service.TestWindowVSCodeService.InitializeAsync(InitializeOptions options, CancellationToken cancellationToken)
unable to create test controller for c# extension: Error: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/6.0.407/vstest.console.dll" /parentprocessid:14107 /port:50800.
```

I have both .NET 6 and .NET 7 SDKs and runtimes installed.

**Comment by GavinPlusPlus** [comment]
Can confirm I am also struggling with this issue on MacOS Ventura 13.4, although the project seems to load just fine on my Windows 11 system. Neither of my systems have spaces anywhere in the file path and both have .net6 and 7.

**Comment by mitch-b** [comment]
I ran into this, and eventually found this failure when inspecting my output logs for "C# Dev Kit - Test Explorer" - 

```bash
Created Test Controller
Using vstest from dotnet sdk in [/usr/local/share/dotnet/sdk/7.0.302].
System.Exception: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
 ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process '/usr/local/bin/dotnet' with working directory '/Users/mitch/src/PROJECT'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   --- End of inner exception stack trace ---
   at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProviders.LocalTestPlatformProvider.ActivateSessionAsync(ITestWindowCallbackService testWindowCallbackService, ITestAgentBridgeCallback TestAgentBridgeCallback, Boolean enableDiagnosticLogging, RuntimeEnvironment runtimeEnvironment, IDictionary`2 environment, IDictionary`2 props, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformAgentPool.<>c__DisplayClass24_0.<<CreateRemoteAgentAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Utilities.EventPumpExtensions.<>c__DisplayClass4_0`1.<<EnqueueAsync>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestWindow.Extensibility.TaskCompletionSourceExtensions.RunTaskWithCatchAsync[T](TaskCompletionSource`1 taskCompletionSource, Func`1 task)
   at Microsoft.VisualStudio.TestWindow.Logging.ILoggerExtensions.CallWithCatchAsync(IInternalLogger log, Func`1 func)
   at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProvider.GetProviderAsync(ITestWindowCallbackService callbackClient, IInternalLogger logger, ITestWindowHostTelemetry telemetryService, ICollection`1 runtimeEnvironments, IDictionary`2 environmentVariables, Boolean enableDiagnosticLogging, Boolean autoInstallRuntimePrereqs, ITestPlatformProviderMap testPlatformProviderMap)
   at Microsoft.VisualStudio.TestWindow.Host.TestWindowStoreService.InitializeVSCodeServiceAsync(InitializeOptions initializeRequest, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestWindow.VSCode.Service.TestWindowVSCodeService.InitializeAsync(InitializeOptions options, CancellationToken cancellationToken)
unable to create test controller for c# extension: Error: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
```

The short story is, I used to run x64 on my M1 mac, but recently shifted to arm64 (after uninstalling dotnet) using the uninstallation tool. Which - by the way - didn't seem to remove the link at `/usr/local/bin/dotnet`. I didn't notice (didn't impact me) until this.

```bash
which dotnet
> /usr/local/share/dotnet/dotnet
sudo ln -sf /usr/local/share/dotnet/dotnet /usr/local/bin/dotnet
```

... relaunched VSCode and everything started working. 

Based on the replies above, this may not help everyone, but could be a bug in the test controller when determining dotnet sdk path.

**Comment by ryandeering** [comment]
That's worked for me @GavinPlusPlus, I believe I had the exact same scenario of accidentally installing the x64 version .NET with the uninstaller script, might be worth flagging there that it might leaving a leftover symlink.

**Comment by GavinPlusPlus** [comment]
> I ran into this, and eventually found this failure when inspecting my output logs for "C# Dev Kit - Test Explorer" -
> 
> ```shell
> Created Test Controller
> Using vstest from dotnet sdk in [/usr/local/share/dotnet/sdk/7.0.302].
> System.Exception: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
>  ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process '/usr/local/bin/dotnet' with working directory '/Users/mitch/src/PROJECT'. No such file or directory
>    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
>    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
>    at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
>    at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
>    --- End of inner exception stack trace ---
>    at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProviders.LocalTestPlatformProvider.ActivateSessionAsync(ITestWindowCallbackService testWindowCallbackService, ITestAgentBridgeCallback TestAgentBridgeCallback, Boolean enableDiagnosticLogging, RuntimeEnvironment runtimeEnvironment, IDictionary`2 environment, IDictionary`2 props, CancellationToken cancellationToken)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformAgentPool.<>c__DisplayClass24_0.<<CreateRemoteAgentAsync>b__0>d.MoveNext()
> --- End of stack trace from previous location ---
>    at Microsoft.VisualStudio.TestWindow.Utilities.EventPumpExtensions.<>c__DisplayClass4_0`1.<<EnqueueAsync>b__1>d.MoveNext()
> --- End of stack trace from previous location ---
>    at Microsoft.VisualStudio.TestWindow.Extensibility.TaskCompletionSourceExtensions.RunTaskWithCatchAsync[T](TaskCompletionSource`1 taskCompletionSource, Func`1 task)
>    at Microsoft.VisualStudio.TestWindow.Logging.ILoggerExtensions.CallWithCatchAsync(IInternalLogger log, Func`1 func)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProvider.GetProviderAsync(ITestWindowCallbackService callbackClient, IInternalLogger logger, ITestWindowHostTelemetry telemetryService, ICollection`1 runtimeEnvironments, IDictionary`2 environmentVariables, Boolean enableDiagnosticLogging, Boolean autoInstallRuntimePrereqs, ITestPlatformProviderMap testPlatformProviderMap)
>    at Microsoft.VisualStudio.TestWindow.Host.TestWindowStoreService.InitializeVSCodeServiceAsync(InitializeOptions initializeRequest, CancellationToken cancellationToken)
>    at Microsoft.VisualStudio.TestWindow.VSCode.Service.TestWindowVSCodeService.InitializeAsync(InitializeOptions options, CancellationToken cancellationToken)
> unable to create test controller for c# extension: Error: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
> ```
> 
> The short story is, I used to run x64 on my M1 mac, but recently shifted to arm64 (after uninstalling dotnet) using the uninstallation tool. Which - by the way - didn't seem to remove the link at `/usr/local/bin/dotnet`. I didn't notice (didn't impact me) until this.
> 
> ```shell
> which dotnet
> > /usr/local/share/dotnet/dotnet
> sudo ln -sf /usr/local/share/dotnet/dotnet /usr/local/bin/dotnet
> ```
> 
> ... relaunched VSCode and everything started working.
> 
> Based on the replies above, this may not help everyone, but could be a bug in the test controller when determining dotnet sdk path.

Hey that fixed it for me! I don't ever remember installing the x64 versions of dotnet but a while back during the Ventura Betas there was a nasty bug with MAUI and I must have installed it by accident. Thanks again, really appreciate it!

**Comment by ydang204** [comment]
> ```shell
> sudo ln -sf /usr/local/share/dotnet/dotnet /usr/local/bin/dotnet
> ```

You save my day! Thank you <3

**Comment by ydang204** [comment]
> I ran into this, and eventually found this failure when inspecting my output logs for "C# Dev Kit - Test Explorer" -
> 
> ```shell
> Created Test Controller
> Using vstest from dotnet sdk in [/usr/local/share/dotnet/sdk/7.0.302].
> System.Exception: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
>  ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process '/usr/local/bin/dotnet' with working directory '/Users/mitch/src/PROJECT'. No such file or directory
>    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
>    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
>    at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
>    at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
>    --- End of inner exception stack trace ---
>    at Microsoft.TestPlatform.VsTestConsole.TranslationLayer.VsTestConsoleProcessManager.StartProcess(ConsoleParameters consoleParameters)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProviders.LocalTestPlatformProvider.ActivateSessionAsync(ITestWindowCallbackService testWindowCallbackService, ITestAgentBridgeCallback TestAgentBridgeCallback, Boolean enableDiagnosticLogging, RuntimeEnvironment runtimeEnvironment, IDictionary`2 environment, IDictionary`2 props, CancellationToken cancellationToken)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformAgentPool.<>c__DisplayClass24_0.<<CreateRemoteAgentAsync>b__0>d.MoveNext()
> --- End of stack trace from previous location ---
>    at Microsoft.VisualStudio.TestWindow.Utilities.EventPumpExtensions.<>c__DisplayClass4_0`1.<<EnqueueAsync>b__1>d.MoveNext()
> --- End of stack trace from previous location ---
>    at Microsoft.VisualStudio.TestWindow.Extensibility.TaskCompletionSourceExtensions.RunTaskWithCatchAsync[T](TaskCompletionSource`1 taskCompletionSource, Func`1 task)
>    at Microsoft.VisualStudio.TestWindow.Logging.ILoggerExtensions.CallWithCatchAsync(IInternalLogger log, Func`1 func)
>    at Microsoft.VisualStudio.TestWindow.Host.TestPlatformProvider.GetProviderAsync(ITestWindowCallbackService callbackClient, IInternalLogger logger, ITestWindowHostTelemetry telemetryService, ICollection`1 runtimeEnvironments, IDictionary`2 environmentVariables, Boolean enableDiagnosticLogging, Boolean autoInstallRuntimePrereqs, ITestPlatformProviderMap testPlatformProviderMap)
>    at Microsoft.VisualStudio.TestWindow.Host.TestWindowStoreService.InitializeVSCodeServiceAsync(InitializeOptions initializeRequest, CancellationToken cancellationToken)
>    at Microsoft.VisualStudio.TestWindow.VSCode.Service.TestWindowVSCodeService.InitializeAsync(InitializeOptions options, CancellationToken cancellationToken)
> unable to create test controller for c# extension: Error: An error occurred when opening the file /usr/local/bin/dotnet with arguments "/usr/local/share/dotnet/sdk/7.0.302/vstest.console.dll" /parentprocessid:51312 /port:51598.
> ```
> 
> The short story is, I used to run x64 on my M1 mac, but recently shifted to arm64 (after uninstalling dotnet) using the uninstallation tool. Which - by the way - didn't seem to remove the link at `/usr/local/bin/dotnet`. I didn't notice (didn't impact me) until this.
> 
> ```shell
> which dotnet
> > /usr/local/share/dotnet/dotnet
> sudo ln -sf /usr/local/share/dotnet/dotnet /usr/local/bin/dotnet
> ```
> 
> ... relaunched VSCode and everything started working.
> 
> Based on the replies above, this may not help everyone, but could be a bug in the test controller when determining dotnet sdk path.

You save my day! Thank you.

**Comment by mhrastegari** [comment]
I cannot load our projects on my `Windows 11` PC either.
And I'm guessing it's because of space in my user folder which is `MH Rastegari`, is there a workaround for it?

![image](https://github.com/microsoft/vscode-dotnettools/assets/42671084/7c2b9935-cfe0-4bfa-bc3d-0d634de36b12)

**Comment by Michael-Eng** [comment]
The space in the user name is this bug and has been fixed.
#24

**Comment by Michael-Eng** [comment]
Space in user name is fixed #24 
The .NET Tools team is looking at making it more obvious that the ARM64 version of .NET tools should be insalled on M1/M2 macs
